src/Common/Common.Api/Reports/ReportsClient.cs

```diff
@@ -93,5 +103,70 @@ public IEnumerable<Import> GetImportsForWorkspace(Guid workspaceId)
         {
             return this.Client.Imports.GetImports(groupId: workspaceId.ToString()).Value?.Select(x => (Import)x);
         }
+
+        public Guid PostImport(string datasetDisplayName, string filePath)
+        {
+            using (var fileStream = new StreamReader(filePath))
+            {
+                var response = this.Client.Imports.PostImportWithFile(
+                    fileStream: fileStream.BaseStream,
+                    datasetDisplayName: datasetDisplayName,
+                    nameConflict: PowerBI.Api.V2.Models.ImportConflictHandlerMode.CreateOrOverwrite
+                );
+                return Guid.Parse(response.Id);
+            }
+        }
+
+        public Guid PostImportForWorkspace(Guid workspaceId, string datasetDisplayName, string filePath)
+        {
+            using (var fileStream = new StreamReader(filePath))
+            {
+                var response = this.Client.Imports.PostImportWithFileInGroup(
+                    groupId: workspaceId.ToString(),
+                    fileStream: fileStream.BaseStream,
+                    datasetDisplayName: datasetDisplayName,
+                    nameConflict: PowerBI.Api.V2.Models.ImportConflictHandlerMode.CreateOrOverwrite
+                );
+                return Guid.Parse(response.Id);
+            }
+        }
+
+        public Report PostReport(string reportName, string filePath)
+        {
+            var importId = this.PostImport(reportName, filePath);
+
+            Import import = null;
+            do
+            {
+                import = this.GetImport(importId: importId);
+                if (import.ImportState != "Succeeded")
+                    System.Threading.Thread.Sleep(500);
+
+            } while (import.ImportState == "Publishing");
+
+            if (import.ImportState != "Succeeded")
+                throw new Exception(string.Format("ImportState is '{0}'", import.ImportState));
+
+            return import.Reports.Single();
+        }
+
+        public Report PostReportForWorkspace(Guid workspaceId, string reportName, string filePath)
+        {
+            var id = this.PostImportForWorkspace(workspaceId, reportName, filePath);
+
+            Import import = null;
+            do
+            {
+                import = this.GetImportForWorkspace(workspaceId: workspaceId, importId: id);
+                if (import.ImportState != "Succeeded")
+                    System.Threading.Thread.Sleep(500);
+
+            } while (import.ImportState == "Publishing");
+
+            if (import.ImportState != "Succeeded")
+                throw new Exception(string.Format("ImportState is '{0}'", import.ImportState));
```

Comment:
It might be helpful to make this exception message a little more descriptive. My suggestion is the following: $"The import was not successful for import ID '{importId}', report name '{reportName}'. The import state was '{import.State}'."